pcdsdevices/interface.py

```diff
@@ -1764,6 +1766,41 @@ def _calc_cache_lightpath_state(self, *args, **kwargs) -> None:
         """
         self.get_lightpath_state(use_cache=False)
 
+    @property
+    def input_branches(self):
+        """
+        return input_branches from happi metadata, unless provided at init
+        """
+        md = getattr(self, 'md', None)
+        if md:
+            return md.get('input_branches')
+        else:
+            return getattr(self, '_input_branches')
```

Comment:
Another way to do this would have been to set `input_branches` and `output_branches` as part of the md setter and drop the leading underscore variants

Reply:
That's true.  I can get rid of some @property-ing here